htsget-config/src/storage/mod.rs

```diff
@@ -0,0 +1,199 @@
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+use crate::resolver::ResolveResponse;
+use crate::storage::local::LocalStorage;
+#[cfg(feature = "s3-storage")]
+use crate::storage::s3::S3Storage;
+use crate::types::{Query, Response, Result};
+
+pub mod local;
+#[cfg(feature = "s3-storage")]
+pub mod s3;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub enum TaggedStorageTypes {
+  #[serde(alias = "local", alias = "LOCAL")]
+  Local,
+  #[cfg(feature = "s3-storage")]
+  #[serde(alias = "s3")]
+  S3,
+}
+
+/// If s3-storage is enabled, then the default is `S3`, otherwise it is `Local`.
```

Comment:
I'm not sure if it's the best idea to have this kind of change in behaviour when the compilation feature flags are different, but I think it is okay.